lib/internal/test_runner/harness.js

```diff
@@ -52,14 +53,16 @@ function createProcessEventHandler(eventName, rootTest) {
       throw err;
     }
 
-    // Check if this error is coming from a test. If it is, fail the test.
-    if (!test || test.finished) {
+    // Check if this error is coming from a test or test hook. If it is, fail the test.
+    if (!test || test.finished || test instanceof TestHook) {
       // If the test is already finished or the resource that created the error
       // is not mapped to a Test, report this as a top level diagnostic.
       let msg;
 
       if (test) {
-        msg = `Warning: Test "${test.name}" generated asynchronous ` +
+        const name = test instanceof TestHook ? `Test Hook "${test.hookType}"` : `Test "${test.name}"`;
+        const relPath = relative(process.cwd(), test.loc.file);
```

Comment:
It is not safe to assume that the test location exists. It won't in cases like the REPL.